lib/cf/CloudControllerClient.js

```diff
@@ -275,6 +278,27 @@ class CloudControllerClient extends HttpClient {
     return this.getResource(`/v2/organizations/${org_guid}`, options);
   }
 
+  getMultiTenantServiceInstanceIdWithName(instance_name, org_guid, space_guid, service_name, options) {
+    return this.getResource(`/v2/spaces/${space_guid}/service_instances?q=name:${instance_name}`, options)
+      .then((serviceInstance) => {
+        if (serviceInstance.resources.length < 1) {
+          logger.debug(`Service instance with name ${instance_name} not found in the space ${space_guid}`);
+          throw new MultiTenantServiceInstanceNameNotFound(instance_name, service_name);
+        }
+        var serviceInstanceId = serviceInstance.resources[0].metadata.guid;
+        return this.findServicePlanByInstanceId(serviceInstanceId)
+          .then((servicePlan) => {
+            var catalogPlan = catalog.getPlan(servicePlan.entity.unique_id);
+            if (catalogPlan.service.name === service_name && catalogPlan.manager.settings.agent !== undefined && catalogPlan.manager.settings.agent.supported_features.indexOf('multi_tenancy') !== -1) {
+              return Promise.resolve(serviceInstanceId);
+            } else {
+              logger.debug(`Service instance with name ${instance_name} was found in the space ${space_guid} but it was not a ${service_name} instance with multi-tenancy feature support.`);
+              throw new MultiTenantServiceInstanceNameNotFound(instance_name, service_name);
```

Comment:
what is the need for this new error type. In the above case you can throw `errors.Forbidden` so semantically it makes more sense to send 403, instead of 404.